player/js/utils/shapes/ZigZagModifier.js

```diff
@@ -0,0 +1,91 @@
+import {
+  extendPrototype,
+} from '../functionExtensions';
+import PropertyFactory from '../PropertyFactory';
+import shapePool from '../pooling/shape_pool';
+import {
+  ShapeModifier,
+} from './ShapeModifiers';
+import { PolynomialBezier } from '../PolynomialBezier';
+
+function ZigZagModifier() {}
+extendPrototype([ShapeModifier], ZigZagModifier);
+ZigZagModifier.prototype.initModifierProperties = function (elem, data) {
+  this.getValue = this.processKeys;
+  this.amplitude = PropertyFactory.getProp(elem, data.s, 0, null, this);
+  this.frequency = PropertyFactory.getProp(elem, data.pt, 0, null, this);
+  this._isAnimated = this.amplitude.effectsSequence.length !== 0 && this.frequency.effectsSequence.length !== 0;
+};
+
+ZigZagModifier.prototype.processPath = function (path, amplitude, frequency) {
+  var pathLength = path._length;
+  var clonedPath = shapePool.newElement();
+  clonedPath.c = path.c;
+  var vX;
+  var vY;
+  var direction = 1;
+
+  if (!path.c) {
+    pathLength -= 1;
+  }
+
+  for (var cur = 0; cur < pathLength; cur += 1) {
+    var bez = PolynomialBezier.shapeSegment(path, cur);
+
+    vX = bez.points[0][0];
+    vY = bez.points[0][1];
+    clonedPath.setTripleAt(vX, vY, vX, vY, vX, vY, clonedPath.length());
+
+    for (var i = 0; i < frequency; i += 1) {
+      var t = (i + 0.5) / frequency;
```

Comment:
`var t = (i + 1) / (frequency + 1);`
This  will split the path into the right amount of segments